pubber/src/main/java/daq/pubber/Pubber.java

```diff
@@ -1017,16 +1038,29 @@ private String extractConfigBlob(String blobName) {
         return null;
       }
       BlobBlobsetConfig blobBlobsetConfig = deviceConfig.blobset.blobs.get(blobName);
-      if (blobBlobsetConfig != null && BlobPhase.FINAL.equals(blobBlobsetConfig.phase)
-          && blobBlobsetConfig.base64 != null) {
-        return new String(Base64.getDecoder().decode(blobBlobsetConfig.base64));
+      if (blobBlobsetConfig != null && BlobPhase.FINAL.equals(blobBlobsetConfig.phase)) {
+        return acquireBlobData(blobBlobsetConfig.url, blobBlobsetConfig.sha256);
       }
       return null;
     } catch (Exception e) {
-      throw new RuntimeException("While extracting config blob " + blobName, e);
+      EndpointConfiguration endpointConfiguration = new EndpointConfiguration();
+      endpointConfiguration.error = e.toString();
+      return JsonUtil.stringify(endpointConfiguration);
     }
   }
 
+  static String acquireBlobData(String url, String sha256) {
+    if (!url.startsWith(DATA_URL_JSON_BASE64)) {
+      throw new RuntimeException("URL encoding not supported: " + url);
+    }
+    byte[] dataBytes = Base64.getDecoder().decode(url.substring(DATA_URL_JSON_BASE64.length()));
```

Comment:
Do we like how this will test if we send malformed data urls etc?

Reply:
Not sure what you mean -- this isn't a test. If there data URL is malformed then it will complain, either through the decoding, or the hash will not match.